### ISSUE TITLE
ENH - Allow using a different distance for the nearest neighbors in fuzzy join

### ISSUE THREAD

**Issue by jeromedockes** [skrub-data/skrub#869]
### Problem Description

ATM we use NearestNeighbors with the l2 distance.
if we could choose the distance to use, then using MinHash as the text encoder and "hamming" as the distance would be an approximation of 1 - Jaccard similarity, which I believed is a common choice for fuzzy joining

### Feature Description

the Joiner would have a "metric" or "distance" parameter that would be forwarded to [NearestNeighbors metric](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestNeighbors.html#sklearn.neighbors.NearestNeighbors)

**Comment by rcap107** [comment]
This is an issue that we might want to look at as part of the Joiners refactoring effort
